File: Algorithms/data-structure/book/bubble-sort.js
/*
bubbleSort(A[], n) 
{
    for last <- n downto 2 {
        sorted <- TRUE;
        for i <- 1 to last-1 {
            if(A[i] > A[i+1]) then {
                A[i] <-> A[i+1];
                sorted <- false;
            }
        }
        if(sorted = TRUE) then return;
    }
}
*/

File: algorithms/bubble_sort.py
def bubble_sort(list):
    # 정렬 완료된 bubble이 마지막에 위치한다.

    unsorted_until_index = len(list) - 1
    sorted = False

    while not sorted:
        sorted = True
        for i in range(unsorted_until_index):
            if list[i] > list[i+1]:
                list[i], list[i+1] = list[i+1], list[i]
                sorted = False
        unsorted_until_index -= 1

    return list


print(bubble_sort([2, 4, 1, 3, 7]))
